io-engine/src/grpc/v1/snapshot.rs

```diff
@@ -384,9 +393,14 @@ impl SnapshotRpc for SnapshotService {
                             &args.snapshot_uuid
                         ) {
                             Some(snap_config) => snap_config,
-                            None => return Err(LvsError::SnapshotConfigFailed {
-                                name: args.replica_uuid,
-                                msg: "tx id / snapshot name not provided".to_string(),
+                            // if any of the prepare parameters not passed,
+                            // return failure as pre-condition failed.
```

Comment:
Should this be invalid argument rather than precondition failed?

Reply:
Indeed. It will be more appropriate. Will change it.